Gems/ROS2/Code/Source/Lidar/LidarSystem.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "LidarSystem.h"
+#include "ROS2/Lidar/LidarRegistrarBus.h"
+
+namespace ROS2
+{
+    AzPhysics::SceneHandle GetPhysicsSceneFromEntityId(const AZ::EntityId& entityId)
+    {
+        auto* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get();
+        auto foundBody = physicsSystem->FindAttachedBodyHandleFromEntityId(entityId);
+        AzPhysics::SceneHandle lidarPhysicsSceneHandle = foundBody.first;
+        if (foundBody.first == AzPhysics::InvalidSceneHandle)
+        {
+            auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+            lidarPhysicsSceneHandle = sceneInterface->GetSceneHandle(AzPhysics::DefaultPhysicsSceneName);
+        }
+
+        AZ_Assert(lidarPhysicsSceneHandle != AzPhysics::InvalidSceneHandle, "Invalid physics scene handle for entity");
+        return lidarPhysicsSceneHandle;
+    }
+
+    void LidarSystem::Activate()
+    {
+        const char* name = "PhysX";
+        const char* description = "Collider-based lidar implementation that uses the PhysX engine's raycasting.";
+        const LidarSystemFeatures supportedFeatures = {
+            /* .m_noise =                   */ false,
+            /* .m_collisionLayers =         */ true,
+            /* .m_entityExclusion =         */ false,
+            /* .m_MaxRangeHitPointConfig =  */ true,
```

Comment:
Thanks for adding these comments, I wish we had C designated initializer support in C++ (apparently 20 has it so maybe one day) - the other option that can sometimes be nicer and easier to read is to use enum bitwise flags.

Reply:
Thanks! The enum bitwise flags are a great idea. Done.